src/web/core/src/main/java/org/geoserver/web/data/workspace/WorkspaceEditPage.java

```diff
@@ -113,6 +113,13 @@ private void init(WorkspaceInfo ws) {
         wsModel = new WorkspaceDetachableModel(ws);
 
         NamespaceInfo ns = getCatalog().getNamespaceByPrefix( ws.getName() );
+
+        if (ns == null) {
+            // unfortunately this may happen if the namespace associated to the workspace was deleted or never created
+            throw new RuntimeException(String.format(
+                    "Workspace '%s' associated namespace doesn't exists.", ws.getName()));
```

Comment:
This will show up as the error message for the user... is there any suggestion we can provide to fix this also? Like I don't know "Go to the workspace page and save again, assuming that will fix the misalignment, or "use the rest api to fix this")

Reply:
If this situation happens it means that the user manually edited the data directory or that the system went in corrupted state. The is nor much the user can do except:

1. Manually try to fix the data directory.
2. Remove the workspace and recreate it.